File: API/Data/Entities/AirConditioner.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Data.Entities;

[Table("air_conditioners")]
public class AirConditioner
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("brand")]
    public string? Brand { get; set; }

    [Column("model")]
    public string? Model { get; set; }

    [Column("kilowatts")]
    public int? Kilowatts { get; set; }

    [Column("description")]
    public string? Description { get; set; }

    [Column("price", TypeName = "numeric(10, 2)")]
    public decimal? Price { get; set; }

    [Column("image_url")]
    public string? ImageUrl { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("pipe_size_liquid")]
    public string? PipeSizeLiquid { get; set; }

    [Column("pipe_size_gas")]
    public string? PipeSizeGas { get; set; }

    [Column("refrigerant_type")]
    public string? RefrigerantType { get; set; }

    [Column("factory_refrigerant_charge")]
    public int? FactoryRefrigerantCharge { get; set; }

    [Column("power_supply_location")]
    public string? PowerSupplyLocation { get; set; }

    [Column("cable_section")]
    public string? CableSection { get; set; }

    [Column("recommended_fuse")]
    public int? RecommendedFuse { get; set; }

    [Column("indoor_dimensions")]
    public string? IndoorDimensions { get; set; }

    [Column("outdoor_dimensions")]
    public string? OutdoorDimensions { get; set; }

    [Column("weight_indoor", TypeName = "decimal(5, 2)")]
    public decimal? WeightIndoor { get; set; }

    [Column("weight_outdoor", TypeName = "decimal(5, 2)")]
    public decimal? WeightOutdoor { get; set; }

    [Column("max_pipe_length")]
    public int? MaxPipeLength { get; set; }

    [Column("max_height_difference")]
    public int? MaxHeightDifference { get; set; }

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }
    
    public ICollection<ErrorCode> ErrorCodes { get; set; } = new HashSet<ErrorCode>();
}

File: frontend/src/types/air-conditioner.ts
export interface AirConditioner {
  id: string;
  name: string;
  brand?: string | null;
  model?: string | null;
  kilowatts?: number | null;
  description?: string | null;
  price?: number | null;
  image_url?: string | null;
  
  pipe_size_liquid?: string | null;
  pipe_size_gas?: string | null;
  max_pipe_length?: number | null;
  max_height_difference?: number | null;
  
  refrigerant_type?: string | null;
  factory_refrigerant_charge?: number | null;
  
  power_supply_location?: string | null;
  cable_section?: string | null;
  recommended_fuse?: number | null;
  
  indoor_dimensions?: string | null;
  outdoor_dimensions?: string | null;
  weight_indoor?: number | null;
  weight_outdoor?: number | null;
}

export interface ErrorCode {
  id: string;
  error_code: string;
  error_name?: string | null;
  description?: string | null;
  air_conditioner_id: string;
  /** Display name of the linked air conditioner, resolved by the backend. */
  air_conditioner_name?: string | null;
  solution?: string | null;
  severity?: string | null;
}

/** Aggregate error-code statistics across the whole catalog (not a single page). */
export interface ErrorCodeStats {
  total: number;
  with_solutions: number;
  air_conditioners: number;
}

export interface CreateAirConditionerRequest {
  name: string;
  brand?: string | null;
  model?: string | null;
  kilowatts?: number | null;
  description?: string | null;
  price?: number | null;
  image_url?: string | null;
  
  pipe_size_liquid?: string | null;
  pipe_size_gas?: string | null;
  max_pipe_length?: number | null;
  max_height_difference?: number | null;
  
  refrigerant_type?: string | null;
  factory_refrigerant_charge?: number | null;
  
  power_supply_location?: string | null;
  cable_section?: string | null;
  recommended_fuse?: number | null;
  
  indoor_dimensions?: string | null;
  outdoor_dimensions?: string | null;
  weight_indoor?: number | null;
  weight_outdoor?: number | null;
}

export interface UpdateAirConditionerRequest extends Partial<CreateAirConditionerRequest> {}

export interface CreateErrorCodeRequest {
  error_code: string;
  error_name?: string | null;
  description?: string | null;
  air_conditioner_id: string;
  solution?: string | null;
  severity?: string | null;
}

export interface UpdateErrorCodeRequest extends Partial<Omit<CreateErrorCodeRequest, 'air_conditioner_id'>> {}


